cmd/conduit/internal/initialize/init.go

```diff
@@ -41,25 +45,34 @@ func formatArrayObject(obj string) string {
 
 }
 
-func runConduitInit(path string, importerFlag string, processorsFlag []string, exporterFlag string) error {
-	var location string
-	if path == "" {
-		path = defaultDataDirectory
-		location = "in the current working directory"
-	} else {
-		location = fmt.Sprintf("at '%s'", path)
+func runConduitInit(path string, configWriter io.Writer, importerFlag string, processorsFlag []string, exporterFlag string) error {
+	if configWriter != nil && path != "" {
+		return errStdoutAndPath
 	}
 
-	if err := os.MkdirAll(path, 0755); err != nil {
-		return err
-	}
+	var location string
 
-	configFilePath := filepath.Join(path, conduit.DefaultConfigName)
-	f, err := os.Create(configFilePath)
-	if err != nil {
-		return fmt.Errorf("runConduitInit(): failed to create %s", configFilePath)
+	// to get here, the path must be initialized
+	if configWriter == nil {
+		if path == "" && configWriter == nil {
```

Comment:
configWriter == nil check is redundant on this line since it's already one in the previous line.